src/app/login/login.component.ts

```diff
@@ -41,6 +42,13 @@ export class LoginComponent {
     iconRegistry: MatIconRegistry,
     sanitizer: DomSanitizer
   ) {
+    if (this.router.url.includes('register')) {
+      this.state = 1;
+    } else {
+      this.state = 0;
+    }
+
+    this.state = window.history.state.data;
```

Comment:
Won't this clobber the value you set in line 46 or 48? It seems like you don't need lines 45-49 if you have this line. Or you don't need this and you should keep lines 45-49.

Reply:
Yes, thank you for catching that. I was testing line 51 beforehand and forgot to delete it.